syntax = "proto3";

package Malloc.chatsystem.net.v1;

service GrpcData {
    rpc Register (User) returns (RegisterResponse);
    rpc sendMessage(stream SendMessageRequest) returns (stream SendMessageResponse);
    }

message SendMessageRequest {
    uint64 timestamp = 1;
    bytes content = 2;
    string from = 3;
    string to = 4;
}

message SendMessageResponse {
    uint64 timestamp = 1;
    bytes content = 2;
    string from = 3;
    string to = 4;
}

message User {
    string name = 1;
    string password = 2;
}

message Users {
    repeated User users = 1;
}

message RegisterResponse {
  bool success = 1;
  string message = 2;
}
